src/app.rs

```diff
@@ -42,7 +42,6 @@ pub async fn build_with_cert(
     let (drain_tx, drain_rx) = drain::channel();
 
     let ready = readiness::Ready::new();
-    let proxy_task = ready.register_task("proxy listeners");
```

Comment:
L50 `workload manager` has registered

Reply:
This is for the proxy listeners are bound which hapens at L75

Reply:
not sure i understand

Reply:
I think there is only one task waiting for workloads discovery

Reply:
This task isn't waiting for workload discovery though, its waiting until we bind the listeners to ports (15008, 15001, etc)

Reply:
Not make much sense to me, if they bind failed, the process will exit.

Reply:
but we don't want to mark as ready before then

Reply:
Understand what you mean, will keep this